docs/creating-modules-for-deployment.md

```diff
@@ -95,31 +96,41 @@ const b = m.contract("B", {
 });
 ```
 
+### Retrieving an artifact
+
+You can use the module system itself to retrieve any of your locally compiled contract artifacts. This allows you to access and use the artifact ABI and bytecode wherever you'd like:
+
+```tsx
+const artifact = m.getArtifact("Foo");
+```
+
 ### Using an existing contract
 
 A user might need to execute a method in a contract that wasn't deployed by Ignition. An existing contract can be leveraged by passing an address and abi:
 
 ```tsx
-const abi = [{...}]
-const uniswap = m.contractAt("UniswapRouter", "0x123...", abi)
+const artifact = m.getArtifact("UniswapRouter");
+const uniswap = m.contractAt("UniswapRouter", "0x123...", artifact.abi)
 
 m.call(uniswap, "swap", { ... })
 ```
 
 ### Deploying from an artifact
 
-To allow you to use your own mechanism for getting the contract artifact, `contract` supports passing an `Artifact` as an optional second parameter:
+Additionally, `contract` supports passing an `Artifact` as an optional second parameter:
 
 ```javascript
-const artifact = hre.artifacts.readArtifactSync("Foo");
```

Comment:
I think we should keep the `hre.artifacts.readArtifactSync("Foo")` as the example; to show it can be done and also because you should just use `m.contract("Foo")` in this case and our docs shouldn't show bad practice, even if it demonstrates the point.